packages/mui-material/src/NativeSelect/NativeSelectInput.test.js

```diff
@@ -111,4 +111,43 @@ describe('<NativeSelectInput />', () => {
       ).to.toHaveComputedStyle(combinedStyle);
     });
   });
+
+  it('it should override with error style when `select` has `error` state', function test() {
```

Comment:
Can you move this test in `Select` component test file (`Select.test.js`) instead? The issue is reported for overriding the style in the MuiSelect component.

Reply:
@ZeeshanTamboli the test was initially placed in `Select.test.js` but the test was failing with indication  that `ownerState` is undefined, even if it works well on an actual usage. Any tip getting to pass?...


<img width="567" alt="image" src="https://user-images.githubusercontent.com/48022904/227787685-2d342a9c-0534-4352-a2ad-56a5c2be6f4c.png">

Reply:
It is because each slot traverses the `styleOverrides`. This may look illogical, even though we have only an `icon` slot in the theme and `SelectIcon` component passes `ownerState`. But this was discussed in #31982 and explained in https://github.com/mui/material-ui/pull/33241#issue-1278357376 that the `styleOverrides` callback is called on each slot, so it is called multiple times. We are going to revise it in v6. To make it work, you can apply the following changes:

```diff
--- a/packages/mui-material/src/Select/Select.js
+++ b/packages/mui-material/src/Select/Select.js
@@ -71,17 +71,17 @@ const Select = React.forwardRef(function Select(inProps, ref) {

   const variant = fcs.variant || variantProp;

+  const ownerState = { ...props, variant, classes: classesProp };
+  const classes = useUtilityClasses(ownerState);
+
   const InputComponent =
     input ||
     {
-      standard: <StyledInput />,
-      outlined: <StyledOutlinedInput label={label} />,
-      filled: <StyledFilledInput />,
+      standard: <StyledInput ownerState={ownerState} />,
+      outlined: <StyledOutlinedInput label={label} ownerState={ownerState} />,
+      filled: <StyledFilledInput ownerState={ownerState} />,
     }[variant];

-  const ownerState = { ...props, variant, classes: classesProp };
-  const classes = useUtilityClasses(ownerState);
-
   const inputComponentRef = useForkRef(ref, InputComponent.ref);

   return (
```